src/averageAges.js

```diff
@@ -55,7 +77,27 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const diffs = [];
+
+  for (const child of people) {
+    const mother = onlyWithSon
+      ? people.find((el) => {
+        if (el.name === child.mother && child.sex === 'm') {
+          return true;
+        }
+
+        return false;
```

Comment:
just pass the condition 
`el.name === child.mother && child.sex === 'm'` converts to true or false any way